aptos-move/block-executor/src/scheduler_status.rs

```diff
@@ -1267,68 +1390,70 @@ mod tests {
             Ordering::Relaxed,
         );
 
-        assert_err!(status.finish_execution(0));
+        assert_err!(statuses.finish_execution(0, 0));
     }
 
     #[test]
     fn remove_stall_err_senarios() {
-        let manager = ExecutionQueueManager::new_for_test(10);
+        let mut statuses =
+            ExecutionStatuses::new_for_test(ExecutionQueueManager::new_for_test(1), vec![
+                ExecutionStatus::new(),
+                ExecutionStatus::new(),
+                ExecutionStatus::new(),
+            ]);
 
         for wrong_shortcut in [DependencyFlag::Executing as u8, 100] {
-            let status = ExecutionStatus::new_for_test(
+            *statuses.get_status_mut(0) = ExecutionStatus::new_for_test(
                 InnerStatus::new_for_test(StatusEnum::Executed, 0),
                 2,
-                &manager,
-                10,
             );
+
             // remove_stall succeeds as it should.
-            assert_ok_eq!(status.remove_stall(), false);
-            assert_eq!(status.num_stalls.load(Ordering::Relaxed), 1);
+            assert_ok_eq!(statuses.remove_stall(0), false);
+            assert_eq!(statuses.get_status(0).num_stalls.load(Ordering::Relaxed), 1);
 
-            status
+            statuses
+                .get_status_mut(0)
                 .dependency_shortcut
                 .store(wrong_shortcut, Ordering::Relaxed);
             // Normal removal that would otherwise succeed should now return an error.
-            assert_err!(status.remove_stall());
+            assert_err!(statuses.remove_stall(0));
         }
 
-        let status = ExecutionStatus::new_for_test(
+        *statuses.get_status_mut(1) = ExecutionStatus::new_for_test(
```

Comment:
Can't we initialize these directly in the vector above, why overwrite?

Reply:
think I fixed this one, I still assign in one test but that seems different case